libphobos/libdruntime/core/thread.d

```diff
@@ -2640,7 +2647,19 @@ private void scanAllTypeImpl( scope ScanAllThreadsTypeFn scan, void* curStackTop
             // would make portability annoying because it only makes sense on Windows.
             scan( ScanType.stack, t.m_reg.ptr, t.m_reg.ptr + t.m_reg.length );
         }
-        version (GNU)
+        version( GNU_EMUTLS )
+        {
+            try
+            {
+                foreach(m; ModuleInfo)
+                {
+                    if(m.scanTLS)
+                        m.scanTLS((p1, p2) => scan(ScanType.tls, p1, p2));
+                }
+            }
+            catch(Exception) {} // Can't throw as long as our delegate doesn't throw
```

Comment:
Can't we just mark the delegate parameter as ``nothrow`` object.d?  Or is some fancy casting required here.